sound/soc/intel/boards/sof_rt5682.c

```diff
@@ -607,7 +607,12 @@ static int sof_audio_probe(struct platform_device *pdev)
 		hdmi_num = 3;
 	}
 
-	dmi_check_system(sof_rt5682_quirk_table);
+	if (dmi_check_system(sof_rt5682_quirk_table) == 0) {
+		/*dmi quirk has no matches */
+		if (pdev->id_entry) {
+			sof_rt5682_quirk = (unsigned long) pdev->id_entry->driver_data;
+		}
+	}
```

Comment:
you will also need a quirk to deal with 4 HDMI outputs on TGL

Reply:
Added a patch with quirk for number of HDMI DAI's. Update hdmi_num as 4 for TGL. Verified on DUT.

Reply:
Added the quirk now.

Reply:
Updated patch set.